src/common/tests/CommonUtilsUT.cpp

```diff
@@ -94,6 +94,19 @@ TEST_F(CommonUtilsTest, LoadStringFromSingleByteFile)
     EXPECT_TRUE(Cleanup(m_path));
 }
 
+TEST_F(CommonUtilsTest, LoadStringFromZeroLengthFile)
```

Comment:
Zero length files worked even before this change. We need a new test case that specifically targets "zero length but non-empty files" (like procfs).

Reply:
Once this will be merged, more PRs will follow to expand the test coverage. For this test case that you suggested in particular, please feel free to propose one via a new PR. unittests are not functional tests and crafting such a file of 0 length but which contains data needs to be done in the test itself, not rely on profcs, unless it is something that works on all target distros.

Reply:
Thanks for your feedback!